Hqub.MusicBrainz/Hqub.MusicBrainz.API/Cache/IRequestCache.cs

```diff
@@ -0,0 +1,33 @@
+﻿
+namespace Hqub.MusicBrainz.API.Cache
+{
+    using System.IO;
+
+    /// <summary>
+    /// A simple cache interface.
+    /// </summary>
+    public interface IRequestCache
+    {
+        /// <summary>
+        /// Add a request and its response to the cache.
+        /// </summary>
+        void Add(string request, Stream response);
+
+        /// <summary>
+        /// Add a request and its response to the cache.
+        /// </summary>
+        /// <returns>True, if a cache entry matching the request was found.</returns>
+        bool TryGetCachedItem(string request, out Stream stream);
+
+        /// <summary>
+        /// Remove all expired entries.
+        /// </summary>
+        /// <returns>The number of removed entries.</returns>
+        int Cleanup();
```

Comment:
Maybe `Cleanup` and `Clear` should be removed from the interface. Though these are common features of a cache, the methods aren't used by the library and the user should be able to decide if the cache implementation should provide those features.